### PR TITLE
Add unit tests for `utils.ts`

### PR DESCRIPTION
**Pull request by mohamedsalem401**
- Add configuration for jest testing
- Add unit tests for all functions in the utils.ts file
- One commented test that should be uncommented or replaced

### CODE REVIEW

**Review by danstarns — CHANGES_REQUESTED** [review]
Looks great, can we add into CI?

File: test/utils.spec.ts [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+            copyRecursiveSync('/src/file.txt', '/dest/file.txt');
+
+            expect(fs.existsSync).toHaveBeenCalledWith('/src/file.txt');
+            expect(fs.statSync).not.toHaveBeenCalled();
+            expect(fs.mkdirSync).not.toHaveBeenCalled();
+            expect(fs.copyFileSync).not.toHaveBeenCalled();
+        });
+    });
+
+    // TODO: Fix this test as we can't mock the process.cwd
```
Comment (danstarns) [inline test/utils.spec.ts]: How come we can't mock?

Comment (mohamedsalem401) [reply test/utils.spec.ts]: I couldn't figure out the solution for this issue. I think it's related to the following issue in the Jest GitHub repository:
[Jest globals differ from Node globals
](https://github.com/jestjs/jest/issues/2549)


Also take a look at:
https://github.com/jestjs/jest/issues/5146

**Comment by mohamedsalem401** [comment]
> Looks great, can we add into CI?

I'll work on that

File: test/utils.spec.ts (hunk 1 above)
Comment (mohamedsalem401) [reply test/utils.spec.ts]: @danstarns  
I've changed the test, it's currently using the same logic as the actual function itself, so it should always pass, but at least we are ensuring that the function is correctly implemented and accessible in our test environment.

```ts
describe('rootResolve', () => {
    it('should resolve a relative path to an absolute path', () => {
        const result = rootResolve('src/index.js');

        expect(result).toBe(path.join(process.cwd(), 'src/index.js'));
    });
});
```

**Review by artf — APPROVED** [review]
🚀

**Pull request merged**